### ISSUE TITLE
unused_async triggers on stubs with todo!()

### ISSUE THREAD

**Issue by trevoranderson** [rust-lang/rust-clippy#15305]
### Summary

Example code that triggers it
```
async fn unimplemented_task() -> Result<(), String> {
    todo!("Implement task");
}
warning: unused `async` for function with no await statements
    = help: consider removing the `async` from this function
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unused_async
    = note: `-W clippy::unused-async` implied by `-W clippy::pedantic`
    = help: to override `-W clippy::pedantic` add `#[allow(clippy::unused_async)]`
```
I expected that `todo!()` or `unimplemented!()` would silence the lint. The fix isn't helpful since the stubbed code will eventually be async, is awaited by other partially implemented code, and has the defacto type of never.

### Reproducer

I tried this code:

```rust
<code>
```

I saw this happen:

```
<output>
```

I expected to see this happen:

### Version

```text

```
